CesiumRasterOverlays/src/UrlTemplateRasterOverlay.cpp

```diff
@@ -0,0 +1,189 @@
+#include <CesiumRasterOverlays/QuadtreeRasterOverlayTileProvider.h>
+#include <CesiumRasterOverlays/UrlTemplateRasterOverlay.h>
+#include <CesiumUtility/Uri.h>
+
+using namespace CesiumAsync;
+using namespace CesiumGeometry;
+using namespace CesiumGeospatial;
+using namespace CesiumUtility;
+
+namespace CesiumRasterOverlays {
+
+namespace {
+static bool caseInsensitiveCompare(std::string_view lhs, std::string_view rhs) {
+  return lhs.size() == rhs.size() &&
+         std::equal(lhs.begin(), lhs.end(), rhs.begin(), [](char a, char b) {
+           return std::tolower(a) == std::tolower(b);
```

Comment:
Dragons. To start with, std::tolower is locale-dependent, so it might not do what we want on all systems. For another, there's this note in the `std::tolower` docs:

> Similarly, they should not be directly used with standard algorithms when the iterator's value type is char or signed char. Instead, convert the value to unsigned char first:

Take a look at `HttpHeaders.h` / `.cpp`, which does case-insensitive comparisons in a way that might still be imperfect, but it should be at least a little more reliable.